### PR TITLE
Fix of constant usage of CameraVideoCapturer in Rtc

### PR DESCRIPTION
**Pull request by gabrysgab**
### Task

 - [JIRA](https://netguru.atlassian.net/browse/BM-555)
 
### Description

Turning on/off the CameraVideoCapturer depending on usage of
VideoPreview and child -> parent streaming.
Refactor of ChildMonitorFragment and its view models communication
Add ServerViewModel tests

### CODE REVIEW

File: app/src/main/res/layout/fragment_child_monitor.xml [hunk 1]
```diff
@@ -1,181 +1,184 @@
...
+                app:activeColor="@color/active_pulsating"
+                app:circleSize="@dimen/spacing_small"
+                app:inactiveColor="@color/inactive_pulsating"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toEndOf="@+id/babyName"
+                app:layout_constraintTop_toTopOf="parent"
+                />
+
+        <ImageButton
```
Comment (netguru-hound) [inline app/src/main/res/layout/fragment_child_monitor.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/4b77b013577f94eab4a3be800c8ff4ab7abae082/app/src/main/res/layout/fragment_child_monitor.xml#L50"></span>Missing <code>contentDescription</code> attribute on image</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/layout/fragment_child_monitor.xml [hunk 2]
```diff
@@ -1,181 +1,184 @@
...
+            android:shadowDy="4"
+            android:shadowRadius="2"
+            android:textSize="16sp"
+            app:layout_constraintEnd_toEndOf="@+id/surfaceView"
+            app:layout_constraintStart_toStartOf="@+id/surfaceView"
+            app:layout_constraintTop_toTopOf="@+id/surfaceView"
+            tools:text="4:59"
+            />
 
     <ImageButton
```
Comment (netguru-hound) [inline app/src/main/res/layout/fragment_child_monitor.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/4b77b013577f94eab4a3be800c8ff4ab7abae082/app/src/main/res/layout/fragment_child_monitor.xml#L134"></span>Missing <code>contentDescription</code> attribute on image</td>
    </tr>
  </tbody>
</table>

File: app/src/main/res/layout/fragment_child_monitor.xml [hunk 3]
```diff
@@ -1,181 +1,184 @@
...
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        tools:visibility="gone"
-        />
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            tools:visibility="gone"
+            />
 
     <ImageView
```
Comment (netguru-hound) [inline app/src/main/res/layout/fragment_child_monitor.xml]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/4b77b013577f94eab4a3be800c8ff4ab7abae082/app/src/main/res/layout/fragment_child_monitor.xml#L173"></span>Missing <code>contentDescription</code> attribute on image</td>
    </tr>
  </tbody>
</table>

**Comment by netguru-hound** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/4b77b013577f94eab4a3be800c8ff4ab7abae082/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcManager.kt#L41">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcManager.kt#L41</a> - This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/netguru/baby-monitor-client-android/blob/4b77b013577f94eab4a3be800c8ff4ab7abae082/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcService.kt#L27">app/src/main/kotlin/co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcService.kt#L27</a> - Prefer using ‘if’ for binary conditions instead of ‘when’.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">:link: <a href="https://netguru.atlassian.net/browse/BM-555">BM-555</a></td>
    </tr>
  </tbody>
</table>

### JaCoCO Code Coverage 4.93% :white_check_mark:
| Class | Covered | Meta | Status |
|:---|:---:|:---:|:---:|
| `co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcService` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/communication/webrtc/WebRtcManager` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/common/SchedulersProvider` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/data/server/CameraState` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/settings/ServerSettingsFragment` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/application/ApplicationModule` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/server/ChildMonitorViewModel` | 0% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/server/ServerViewModel` | 64% | 0% | :white_check_mark: |
| `co/netguru/baby/monitor/client/feature/server/ChildMonitorFragment` | 0% | 0% | :white_check_mark: |


#### APK file size

| size |
| --- |
| 22.1MB |



#### Number of method references

| Dex file | count |
| --- | --- |
| classes2.dex | 22565 |
| classes.dex | 22527 |



<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ServerViewModel.kt [hunk 4]
```diff
@@ -87,16 +97,43 @@ class ServerViewModel @Inject constructor(
...
-            .observeOn(AndroidSchedulers.mainThread())
+    fun handleRtcServerConnectionState(webRtcServiceBinder: WebRtcService.Binder) {
+        compositeDisposable += webRtcServiceBinder.getConnectionObservable()
+            .subscribeOn(schedulersProvider.io())
+            .observeOn(schedulersProvider.mainThread())
             .subscribeBy(
-                onNext = { mutableRtcConnectionStatus.postValue(it) },
+                onNext = {
+                    handleStreamState(it)
+                    mutableRtcConnectionStatus.postValue(it)
```
Comment (kkogut95) [inline app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ServerViewModel.kt]: You are observing it on mainThread, shouldn't you set value with setter `mutableRtcConnectionStatus.value = it`? 🤔

Comment (gabrysgab) [reply app/src/main/kotlin/co/netguru/baby/monitor/client/feature/server/ServerViewModel.kt]: Yeah the  .observeOn(schedulersProvider.mainThread()) is quite obsolete here if everything is being set using postValue which does it on mainThread anyway. I'll change it

**Pull request merged**